src/main/java/duke/Storage.java

```diff
@@ -34,20 +34,32 @@ public Storage(String pathname) {
     public void load(TaskList taskList) throws IOException, MissingInfoException {
         try {
             File directory = new File("data");
+            checkDirectory(directory);
 
-            if (!directory.exists()) {
-                directory.mkdir();
-            }
+            File file = new File(this.pathname);
```

Comment:
I think this could be an unnecessary use of the "this" keyword. In general, try to use the "this" only when a field is shadowed by a method or constructor parameter.